routers/web/auth/oauth2_provider.go

```diff
@@ -104,7 +104,8 @@ func InfoOAuth(ctx *context.Context) {
 		Picture:  ctx.Doer.AvatarLink(ctx),
 	}
 
-	groups, err := oauth2_provider.GetOAuthGroupsForUser(ctx, ctx.Doer)
+	onlyPublicGroups := ctx.IsSigned && (ctx.FormString("private") == "" || ctx.FormBool("private"))
```

Comment:
ctx.FormString("private") == "" || ctx.FormBool("private")
It seems it's not right here? Maybe `ctx.FormString("private") == "" || !ctx.FormBool("private")`.